src/main/java/spoon/reflect/factory/CompilationUnitFactory.java

```diff
@@ -76,6 +79,16 @@ public CompilationUnit create(String filePath) {
 	}
 
 	/**
+	 * Removes compilation unit from the cache and returns it
+	 * Used by JDTSnippetCompiler to remove processed snippet from the cache
+	 * @param filePath
+	 * @return a cached compilation unit or null
+	 */
+	public CompilationUnit remove(String filePath) {
```

Comment:
> removeFromCache

good idea

> package-visible

It is different package ... or I do not know this java feature...